### ISSUE TITLE
Support generating examples from individual properties

### ISSUE THREAD

**Issue by Stranger6667** [kiwicom/schemathesis#467]
Schema example:

```
openapi: "3.0.3"
info:
  description: desc
  title: title
  version: "1.0"
servers:
  - url: http://localhost:5000

paths:
  /products/{product_code}:
    parameters:
      - name: product_code
        in: path
        required: true
        schema:
          type: string
          example: code

    put:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Product'
      responses:
        201:
          description: d
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Product'

components:
  securitySchemes:
    hawk:
      type: http
      scheme: bearer
      bearerFormat: HAWK
  schemas:
    Product:
      type: object
      properties:
        name:
          type: string
          example:
            name
        verify_mac:
          type: boolean
          example:
            True
        mac_ranges:
          type: array
          items:
            type: object
          example:
            [
              {"start": "000413230000", "end": "00041323ffff"}
            ]
        default_settings:
          type: object
        groups:
          type: array
          items:
            type: string
            format: uuid
      required:
        - name
        - default_settings
        - groups
        - mac_ranges
        - verify_mac


security:
  - hawk: []
```

In a generic case, examples can be specified on different levels inside the schema, which means that to get a concrete example payload all schema should be traversed and examples should be combined.

Besides that, it may have that a parent schema specifies an example, but also children specify them too - in this case multiple examples should be generated in the end.

**Comment by jonahbenton** [comment]
+1 for this. 

For simple cases- a flat standalone schema object with a bunch of properties- is there an easy way to coax object generation?

**Comment by Stranger6667** [comment]
The simplest move we can do at the moment is to add a hook for example generation - I'll create a separate issue for that, it will provide a way to customize examples, but will take much less time than implementing a full-pledged solution

**Comment by Stranger6667** [comment]
In `1.6.0` I added a new hook, that provides a way to manipulate examples.

https://schemathesis.readthedocs.io/en/stable/customization.html#before-add-examples

Let me know if it is sufficient for your case (I hope to implement a fully automated way of examples parsing soon)

**Comment by barrett-schonefeld** [comment]
@Stranger6667 I will work on this issue!

**Comment by Stranger6667** [comment]
Great! Let me know if I can help

**Comment by Stranger6667** [comment]
Resolved in #637 

I will release this feature this / next week
